src/metadata/api.ts

```diff
@@ -1,28 +1,16 @@
-import {
-  createAppDataDoc,
-  createQuoteMetadata,
-  createReferrerMetadata,
-  createOrderClassMetadata,
-  latest,
-  getAppDataSchema,
-  validateAppDataDoc,
-} from '@cowprotocol/app-data'
-import log from 'loglevel'
-import { Context } from '../../utils/context'
-import { getSerializedCID, loadIpfsFromCid } from '../../utils/appData'
-import { calculateIpfsCidV0, pinJSONToIPFS } from '../../utils/ipfs'
+import { getSerializedCID, loadIpfsFromCid } from './utils/appData'
+import { calculateIpfsCidV0, pinJSONToIPFS } from './utils/ipfs'
 import { AnyAppDataDocVersion, LatestAppDataDocVersion, IpfsHashInfo, GenerateAppDataDocParams } from './types'
-import { CowError } from '../../utils/common'
+import { CowError } from '../common/cow-error'
+import { IpfsConfig } from '../common/configs'
 
 const DEFAULT_APP_CODE = 'CowSwap'
+const REFERRER_VERSION = '0.1.0'
+const QUOTE_VERSION = '0.2.0'
+const ORDER_CLASS_VERSION = '0.1.0'
+const APP_DATA_VERSION = '0.5.0'
```

Comment:
I don't like this part because we are hardcoding individual appData doc versions in the SDK.
The point of the appData package is to abstract this away from the SDK

One can argue that the SDK must be updated anyway when appData changes and that's true, but hard coding the version here makes it one more place to update and to be forgotten.

Reply:
Discussed, moved back the previous approach with lazy loading